File: python/oelMVCS/Core/Board.py
# -*- coding: utf-8 -*-

class Board:
    """内部通许主板"""
    def __init__(self):
        self.__config: Config = None
        self.__logger: Logger = None
        self.__modelCenter: ModelCenter = None
        self.__viewCenter: ViewCenter = None
        self.__controllerCenter: ControllerCenter = None
        self.__serviceCenter: ServiceCenter = None

    def getConfig() -> Config:
        return self.__config

    def getLogger() -> Logger:
        return self.__logger

    def getModelCenter() -> ModelCenter:
        return self.__modelCenter

    def getViewCenter() -> ViewCenter:
        return self.__viewCenter

    def getControllerCenter() -> ControllerCenter:
        return self.__controllerCenter

    def getServiceCenter() -> ServiceCenter:
        return self.__serviceCenter

    def setConfig(_value:Config):
        self.__config = _value

    def setLogger(_value:Logger):
        self.__logger = _value

    def setModelCenter(_value:ModelCenter):
        self.__modelCenter = _value

    def setViewCenter(_value:ViewCenter):
        self.__viewCenter = _value

    def setControllerCenter(_value:ControllerCenter):
        self.__controllerCenter = _value

    def setServiceCenter(_value:ServiceCenter):
        self.__serviceCenter = _value

File: csharp/oelMVCS/Core/Board.cs

/********************************************************************
     Copyright (c) XTechCloud
     All rights reserved.
*********************************************************************/

namespace XTC.oelMVCS
{
    // 内部通讯的主板
    // get和set方法为多语言规范，不直接使用C#的属性存取
    internal class Board
    {
        internal void setServiceCenter(ServiceCenter? _value)
        {
            serviceCenter_ = _value;
        }

        internal void setControllerCenter(ControllerCenter? _value)
        {
            controllerCenter_ = _value;
        }

        internal void setViewCenter(ViewCenter? _value)
        {
            viewCenter_ = _value;
        }

        internal void setModelCenter(ModelCenter? _value)
        {
            modelCenter_ = _value;
        }

        public void setConfig(Config? _value)
        {
            config_ = _value;
        }

        public void setLogger(Logger? _logger)
        {
            logger_ = _logger;
        }

        public Config? getConfig()
        {
            return config_;
        }

        public Logger? getLogger()
        {
            return logger_;
        }

        internal ServiceCenter getServiceCenter()
        {
            return serviceCenter_!;
        }

        internal ControllerCenter getControllerCenter()
        {
            return controllerCenter_!;
        }

        internal ViewCenter getViewCenter()
        {
            return viewCenter_!;
        }

        internal ModelCenter getModelCenter()
        {
            return modelCenter_!;
        }

        private ServiceCenter? serviceCenter_;

        private ControllerCenter? controllerCenter_;

        private ViewCenter? viewCenter_;

        private ModelCenter? modelCenter_;

        private Config? config_;

        private Logger? logger_;
    }
}
